From 7ed8f5afbe42d043dc63104e3b1606bd1d1396a1 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 31 Oct 2024 16:32:00 +0100
Subject: [PATCH 1448/1695] drivers: firewall: stm32_etzpc: explicit index in
 DECPROT string names

Explicit the indices assigned to DECPROT helper string names.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 32f11c59047f7f8e108ac8d4e7f4799b0aee85d4 drivers: firewall: stm32_etzpc: explicit index in DECPROT string names]
Change-Id: I6292ae9dfbbfc53352e10dd06c8f755b9f197bc0
Link: https://github.com/OP-TEE/optee_os/commit/d735136fcc0b3ccaf43c422c3d32398279ecef34
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/419549
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/drivers/firewall/stm32_etzpc.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/core/drivers/firewall/stm32_etzpc.c b/core/drivers/firewall/stm32_etzpc.c
index bb9da7c4a..40c14ef5d 100644
--- a/core/drivers/firewall/stm32_etzpc.c
+++ b/core/drivers/firewall/stm32_etzpc.c
@@ -106,10 +106,10 @@ struct etzpc_device {
 };
 
 static const char *const etzpc_decprot_strings[] __maybe_unused = {
-	"ETZPC_DECPROT_S_RW",
-	"ETZPC_DECPROT_NS_R_S_W",
-	"ETZPC_DECPROT_MCU_ISOLATION",
-	"ETZPC_DECPROT_NS_RW",
+	[ETZPC_DECPROT_S_RW] = "ETZPC_DECPROT_S_RW",
+	[ETZPC_DECPROT_NS_R_S_W] = "ETZPC_DECPROT_NS_R_S_W",
+	[ETZPC_DECPROT_MCU_ISOLATION] = "ETZPC_DECPROT_MCU_ISOLATION",
+	[ETZPC_DECPROT_NS_RW] = "ETZPC_DECPROT_NS_RW",
 };
 
 /* Temporary firewall controller reference */
-- 
2.47.2

